### PR TITLE
Implement More psr7 methods

### PR DESCRIPTION
**Pull request by markstory**
Build out more of the PSR7 interface. I've continued to have methods access the existing public properties to help ensure wrinkle-free inter-operation between the new and existing method sets.

Next I'll be working on uploaded files, and then headers.

### CODE REVIEW

File: src/Network/Request.php [hunk 1]
```diff
@@ -961,6 +961,17 @@ public function header($name)
 
     /**
      * Get the HTTP method used for this request.
+     *
+     * @return string The name of the HTTP method used.
+     * @deprected 3.4.0 This method will be removed in 4.0.0. Use getMethod() instead.
```
Comment (josegonzalez) [inline src/Network/Request.php]: `deprecated`.

Comment (markstory) [inline src/Network/Request.php]: Thanks.

**Comment by markstory** [comment]
This change is part of #9325

**Comment by ADmad** [comment]
LGTM

**Pull request merged**